src/debug/telemetry/telemetry.c

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2023 Intel Corporation. All rights reserved.
+//
+// Author: Tobiasz Dryjanski <tobiaszx.dryjanski@intel.com>
+
+#include <errno.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <adsp_debug_window.h>
+#include <mem_window.h>
+#include <zephyr/debug/sparse.h>
+#include <sof/debug/telemetry/telemetry.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include "adsp_debug_window.h"
+
+/* systic vars */
+int systick_counter;
+int prev_ccount;
+int perf_period_sum_;
+int perf_period_cnt_;
```

Comment:
If i remember correctly, each core has an instance of an LL scheduler, right? Those variables in original firmware were in CoreServices class, which I'm assuming was there to provide some storage and functionality per core. As those variables here are used in update_telemetry function, if there are more LL schedulers running, can be interruped and vars can have their value changed. I'm guessing in this case providing one set of those per core should fix the problem then?

Reply:
yes, one per core and uncached, which is also the default for static writable data